server/src/platform/drbg/helpers.rs

```diff
@@ -0,0 +1,52 @@
+use super::Drbg;
+
+/// Implement `fill_bytes` via `next_u64` and `next_u32`, little-endian order.
+///
+/// The fastest way to fill a slice is usually to work as long as possible with
+/// integers. That is why this method mostly uses `next_u64`, and only when
+/// there are 4 or less bytes remaining at the end of the slice it uses
+/// `next_u32` once.
+pub fn fill_bytes_via_next<R: Drbg>(
+    rng: &mut R,
+    additional_input: &[u8],
+    dest: &mut [u8],
+) -> Result<(), R::Error> {
+    let mut left = dest;
+    while left.len() >= 8 {
+        let (l, r) = { left }.split_at_mut(8);
+        left = r;
+        let chunk: [u8; 8] = rng.next_u64(additional_input)?.to_le_bytes();
```

Comment:
This construct may potentially break NIST.SP.800.90Ar1 compliance. The "additional input" is typically only allowed to be added to the internal DRBG state before and/or after the actual pseudo random data generation.

For example in spec the Hash_DRBG_Generate process will mix the "additional input" into the internal DRBG state  V - and then a call to Hashgen, which works on the state V to generate up to 2¹⁹ bits.

The  HMAC_DRBG_Generate will transform the internal DRBG state using "additional input" only before and after the generation loop.